File: src/test/java/ru/romanow/skip/SkipMethodAspect.java
package ru.romanow.skip;

import org.aspectj.lang.ProceedingJoinPoint;
import org.aspectj.lang.annotation.Around;
import org.aspectj.lang.annotation.Aspect;
import org.aspectj.lang.annotation.Pointcut;

@Aspect
public class SkipMethodAspect {

    @Pointcut("@annotation(ru.romanow.skip.SkipMethod)")
    public void withStepAnnotation() {}

    @Pointcut("execution(* *(..))")
    public void anyMethod() {}

    @Around("anyMethod() && withStepAnnotation()")
    public DummyObject methodAround(final ProceedingJoinPoint joinPoint) {
        return DummyObject.buildDefault();
    }
}

File: src/test/java/ru/romanow/skip/SkipMethodAspect.kt
package ru.romanow.skip

import org.aspectj.lang.ProceedingJoinPoint
import org.aspectj.lang.annotation.Around
import org.aspectj.lang.annotation.Aspect
import org.aspectj.lang.annotation.Pointcut
import org.aspectj.lang.reflect.MethodSignature

@Aspect
class SkipMethodAspect {
    private val skipOn = System.getProperty("skipOn")

    @Pointcut("@annotation(ru.romanow.skip.SkipMethod)")
    fun withStepAnnotation() {
    }

    @Pointcut("execution(* *(..))")
    fun anyMethod() {
    }

    @Around("anyMethod() && withStepAnnotation()")
    fun methodAround(joinPoint: ProceedingJoinPoint): DummyObject {
        var environment: SkipMethod.Environments? = null
        if (!skipOn.isNullOrEmpty()) {
            environment = SkipMethod.Environments.valueOf(skipOn)
        }

        val methodSignature = joinPoint.signature as MethodSignature
        val skipMethod = methodSignature.method.getAnnotation(SkipMethod::class.java)
        return if (environment in skipMethod.skipOn) {
            buildDefault()
        } else {
            joinPoint.proceed() as DummyObject
        }
    }
}